### DISCUSSION TITLE
How can I use a Avalonia control from Nuget?

### DISCUSSION THREAD

**Discussion by ijklam** [fabulous-dev/Fabulous.Avalonia#217] [Q&A]
How can I use a Avalonia control from Nuget in Fabulous.Avalonia?

**Comment by edgarfgp** [comment]
@Tangent-90 Any control in particular that you want to use in Fabulous ?

**Reply by ConradRoesch** [reply]
Since we've not heard back on a particular control, I'll propose one:
https://www.nuget.org/packages/Mapsui.Avalonia
I chose this example since maps are required by all sorts of applications and are not included in the base AvaloniaUI framework.

I suppose that the proper way to support this would be creating a Mapsui.Fabulous.Avalonia NuGet library that wraps the standard Mapsui.Avalonia library, using the WidgetBuilder approach that exposes the standard AvaloniaUI controls.

But perhaps there is an easier way to incorporate an Avalonia NuGet control like this out-of-the-box that I am overlooking?

**Reply by edgarfgp** [reply]
@ConradRoesch `Mapsui` seems like a good control to use for Maps.  For this non official control we have normally 2 options.

1. We encourage people to create its own `Fabulous.Avalonia` control wrapper in their individual GitHub profile like I tried to do with [AvaloniaEdit](https://github.com/edgarfgp/Fabulous.AvaloniaEdit) using a naming convention like `Fabulous.Avalonia.AvaloniaEdit`
2. We can host it in the Fabulous GitHub organization like we have done for [Fabulous.MauiControl.Maps](https://github.com/fabulous-dev/Fabulous.MauiControls.Maps) in the past and give maintainer permissions so we can help/assist maintaining, releasing etc. if it is necessary.

If you are interested let me know :)

**Reply by ConradRoesch** [reply]
I'll take a stab at this when I find some time. I'll start on my GitHub profile, and If I get far enough along that it could be broadly useful, I'd be happy to move it to Fabulous if the organization would like to host it.
